src/LttngNetHelpers.c

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 199309L
+#endif
+
+#include <lttngh/LttngHelpers.h>
+#include <arpa/inet.h>
+#include <stdio.h> // sprintf
+
+void lttngh_FormatIPv4(const void* pIPv4, char* buf16)
+{
+    inet_ntop(AF_INET, pIPv4, buf16, LTTNGH_FORMAT_IPV4_LEN);
+    assert(strlen(buf16) < LTTNGH_FORMAT_IPV4_LEN);
+    buf16[LTTNGH_FORMAT_IPV4_LEN - 1] = 0;
+}
+
+void lttngh_FormatIPv6(const void* pIPv6, char* buf46)
+{
+    inet_ntop(AF_INET6, pIPv6, buf46, LTTNGH_FORMAT_IPV6_LEN);
+    assert(strlen(buf46) < LTTNGH_FORMAT_IPV6_LEN);
+    buf46[LTTNGH_FORMAT_IPV6_LEN - 1] = 0;
+}
+
+void lttngh_FormatSockaddr(const void* pSockaddr, unsigned cbSockaddr,

```

Comment:
Note: On Windows, it is typical for decoders to support sockaddrs for IPv4, IPv6, and Ethernet. This implementation supports sockaddrs for IPv4 and IPv6 but not Ethernet. If somebody can tell me what AF constant and sockaddr struct is normally used for Linux, it would be easy to add. I saw something about `AF_PACKET`, but without confirmation, better to leave it unsupported.